src/springfield/fields.py

```diff
@@ -390,12 +390,26 @@ def __init__(self, entity, *args, **kwargs):
                        Use 'self' to use the :class:`Entity` class that
                        this field is already bound to.
         """
-        self.type = entity
+        self._type = entity
         super(EntityField, self).__init__(*args, **kwargs)
 
+    @property
+    def type(self):
```

Comment:
Maybe you could do `cls._type = imported_class` so it's only done once per class instead of once per instance.